ui/src/shared/components/ThresholdsList.tsx

```diff
@@ -1,34 +1,101 @@
-import React, {Component} from 'react'
-import PropTypes from 'prop-types'
+import React, {PureComponent} from 'react'
 import {connect} from 'react-redux'
 import {bindActionCreators} from 'redux'
 
 import _ from 'lodash'
 import uuid from 'uuid'
 
 import Threshold from 'src/dashboards/components/Threshold'
-import ColorDropdown from 'shared/components/ColorDropdown'
+import ColorDropdown from 'src/shared/components/ColorDropdown'
 
 import {updateThresholdsListColors} from 'src/dashboards/actions/cellEditorOverlay'
-import {colorsNumberSchema} from 'shared/schemas'
+import {ColorNumber} from 'src/types/colors'
 
 import {
   THRESHOLD_COLORS,
   DEFAULT_VALUE_MIN,
   DEFAULT_VALUE_MAX,
   MAX_THRESHOLDS,
   THRESHOLD_TYPE_BASE,
-} from 'shared/constants/thresholds'
+} from 'src/shared/constants/thresholds'
 import {ErrorHandling} from 'src/shared/decorators/errors'
 
 const formatColor = color => {
   const {hex, name} = color
   return {hex, name}
 }
 
+interface Props {
+  onResetFocus: () => void
+  showListHeading: boolean
+  thresholdsListType: string
+  thresholdsListColors: ColorNumber[]
+  handleUpdateThresholdsListColors: (c: ColorNumber[]) => void
+}
+
 @ErrorHandling
-class ThresholdsList extends Component {
-  handleAddThreshold = () => {
+class ThresholdsList extends PureComponent<Props> {
+  public static defaultProps: Partial<Props> = {
+    showListHeading: false,
+  }
+
+  get sortedColors() {
```

Comment:
This should probably be a `private` getter

Reply:
and seems like we might also want a type on the output? not as sure about typing the input on a getter, but i suppose once we flip the "no implicit" switch, we'll have to anyway?